src/TLuaInterpreter.cpp

```diff
@@ -15592,3 +15597,227 @@ int TLuaInterpreter::disableClickthrough(lua_State* L)
     mudlet::self()->setClickthrough(&host, windowName, false);
     return 0;
 }
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Text_Functions#addWordToDictionary
+int TLuaInterpreter::addWordToDictionary(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    if (!host.mEnableSpellCheck) {
+        lua_pushnil(L);
+        lua_pushstring(L, "spell-checking is not enabled in the preferences for this profile");
```

Comment:
Is this an architectural limitation? Because otherwise I don't see why should the Lua API be off when the user has disabled spellchecking.

Reply:
This seems like a least surprising option as far as I see it!

if they *have* switched off spellchecking they probably have not bothered (or cannot) set a reasonable dictionary for their locale (e.g. China) therefore it seems reasonable to switch it off here as well - and handle requests in reasonable run-time behaviour (i.e. with a nil and a tell them why message).

Also bear in mind that I will be reverting the change in that option so that it is enabled (the language will be ser for the "English (American)" which we bundle by default). So this will be an issue only for someone who has turned it off themselves anyway.

Reply:
> Also bear in mind that I will be reverting the change in that option so that it is enabled (the language will be ser for the "English (American)" which we bundle by default)

Actually it seems as if it is not turned off even though I thought I had done so. Ah well maybe I had undone something already. :man_shrugging: :

Reply:
It turns out that the control was unchecked so as to show that when there is not a profile loaded - it takes on the current setting of the `Host` instance when that is loaded in. And **that** does default to being enabled...

Reply:
I think of the API and Mudlets behaviour as separate - you should be able to use it while the preference is off. It's like we have a limitation where you can't use any of the map functions while the map is off.

Reply:
Okay, that is now in place.